Website/src/rendering/src/components/Products/ProductQuantityInput.tsx

```diff
@@ -0,0 +1,57 @@
+import { PriceSchedule, RequiredDeep } from 'ordercloud-javascript-sdk';
+import { ChangeEvent, FunctionComponent } from 'react';
+
+interface ProductQuantityInputProps {
+  controlId: string;
+  priceSchedule: RequiredDeep<PriceSchedule>;
+  label?: string;
+  disabled?: boolean;
+  quantity: number;
+  onChange: (quantity: number) => void;
+}
+
+const ProductQuantityInput: FunctionComponent<ProductQuantityInputProps> = ({
+  controlId,
+  priceSchedule,
+  label = 'Quantity',
+  disabled,
+  quantity,
+  onChange,
+}) => {
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    onChange(Number(e.target.value));
+  };
+
+  const handleSelectChange = (e: ChangeEvent<HTMLSelectElement>) => {
+    onChange(Number(e.target.value));
+  };
+
+  return (
+    <label htmlFor={controlId}>
+      {label}
+      {priceSchedule.RestrictedQuantity ? (
```

Comment:
This line is another example of a condition inside the main JSX of the component.